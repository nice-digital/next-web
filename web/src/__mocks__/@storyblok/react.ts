export const storyblokInit = (): null => null;

export const apiPlugin = (): null => null;
