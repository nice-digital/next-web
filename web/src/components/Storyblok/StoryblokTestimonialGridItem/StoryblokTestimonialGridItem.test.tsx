/* eslint-disable testing-library/no-node-access */
import { render } from "@testing-library/react";

import {
	StoryblokTestimonialGridItem,
	StoryblokTestimonialGridItemProps,
} from "./StoryblokTestimonialGridItem";

const mockStoryBlokTestimonialGridItem: StoryblokTestimonialGridItemProps = {
	blok: {
		image: {
			id: 123456789,
			name: "Kittens mate.",
			filename: "https://placekitten.com/408/287",
			alt: null,
			fieldtype: "asset",
			title: null,
			focus: null,
		},
		quoteName: "Test Name",
		quoteRole: "Test Role",
		quoteText: "Test Quote Text",
		variant: "transparent",
		_uid: "",
		component: "testimonialGridItem",
	},
};

const mockStoryBlokTestimonialGridItemTransparent: StoryblokTestimonialGridItemProps =
	{
		blok: {
			...mockStoryBlokTestimonialGridItem.blok,
			variant: "transparent",
		},
	};

describe("Storyblok testimonial Default component", () => {
	it("should match snapshot for default variant", () => {
		const { container } = render(
			<StoryblokTestimonialGridItem {...mockStoryBlokTestimonialGridItem} />
		);
		expect(container).toMatchSnapshot();
	});

	it("should match snapshot for the default transparent variant", () => {
		const { container } = render(
			<StoryblokTestimonialGridItem
				{...mockStoryBlokTestimonialGridItemTransparent}
			/>
		);
		expect(container).toMatchSnapshot();
	});
});
