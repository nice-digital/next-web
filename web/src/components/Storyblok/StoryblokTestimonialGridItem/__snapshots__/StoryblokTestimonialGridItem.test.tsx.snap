// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyblok testimonial Default component should match snapshot for default variant 1`] = `
<div>
  <div
    class="testimonial testimonial--default testimonial-max-width"
    data-component="testimonial--default"
    data-testid="testimonial-default"
    data-tracking="testimonial"
  >
    <div
      class="testimonial__mainContent"
    >
      <figure
        class="testimonial__content"
      >
        <blockquote
          class="testimonial__quote"
        >
          Test Quote Text
        </blockquote>
        <figcaption
          class="testimonial__person"
        >
          <picture>
            <source
              srcset="https://placekitten.com/408/287/m/328x328/filters:format%28avif%29:quality%2880%29"
              type="image/avif"
            />
            <source
              srcset="https://placekitten.com/408/287/m/328x328/filters:format%28webp%29:quality%2880%29"
              type="image/webp"
            />
            <img
              alt="some alt text"
              class="testimonial__image testimonial__imageMobile"
              data-nimg="1"
              decoding="async"
              height="328"
              loading="lazy"
              src="/_next/image?url=https%3A%2F%2Fplacekitten.com%2F408%2F287%2Fm%2F328x328%2Ffilters%3Aformat%2528jpeg%2529%3Aquality%252880%2529&w=750&q=75"
              srcset="/_next/image?url=https%3A%2F%2Fplacekitten.com%2F408%2F287%2Fm%2F328x328%2Ffilters%3Aformat%2528jpeg%2529%3Aquality%252880%2529&w=384&q=75 1x, /_next/image?url=https%3A%2F%2Fplacekitten.com%2F408%2F287%2Fm%2F328x328%2Ffilters%3Aformat%2528jpeg%2529%3Aquality%252880%2529&w=750&q=75 2x"
              style="color: transparent; max-width: 100%; height: auto;"
              width="328"
            />
          </picture>
          <p
            class="testimonial__details"
          >
            <span
              class="testimonial__name"
            >
              Test Name
            </span>
            <span
              class="testimonial__job"
            >
              Test Role
            </span>
          </p>
        </figcaption>
      </figure>
      <div
        class="testimonial__mainImageContainer"
      >
        <picture>
          <source
            srcset="https://placekitten.com/408/287/m/328x328/filters:format%28avif%29:quality%2880%29"
            type="image/avif"
          />
          <source
            srcset="https://placekitten.com/408/287/m/328x328/filters:format%28webp%29:quality%2880%29"
            type="image/webp"
          />
          <img
            alt="some alt text"
            class="testimonial__image"
            data-nimg="1"
            decoding="async"
            height="328"
            loading="lazy"
            src="/_next/image?url=https%3A%2F%2Fplacekitten.com%2F408%2F287%2Fm%2F328x328%2Ffilters%3Aformat%2528jpeg%2529%3Aquality%252880%2529&w=750&q=75"
            srcset="/_next/image?url=https%3A%2F%2Fplacekitten.com%2F408%2F287%2Fm%2F328x328%2Ffilters%3Aformat%2528jpeg%2529%3Aquality%252880%2529&w=384&q=75 1x, /_next/image?url=https%3A%2F%2Fplacekitten.com%2F408%2F287%2Fm%2F328x328%2Ffilters%3Aformat%2528jpeg%2529%3Aquality%252880%2529&w=750&q=75 2x"
            style="color: transparent; max-width: 100%; height: auto;"
            width="328"
          />
        </picture>
      </div>
    </div>
  </div>
</div>
`;

exports[`Storyblok testimonial Default component should match snapshot for the default transparent variant 1`] = `
<div>
  <div
    class="testimonial testimonial--transparent testimonial-max-width"
    data-component="testimonial--transparent"
    data-testid="testimonial-transparent"
    data-tracking="testimonial"
  >
    <div
      class="testimonial__mainContent"
    >
      <figure
        class="testimonial__content"
      >
        <blockquote
          class="testimonial__quote"
        >
          Test Quote Text
        </blockquote>
        <figcaption
          class="testimonial__person"
        >
          <picture>
            <source
              srcset="https://placekitten.com/408/287/m/328x328/filters:format%28avif%29:quality%2880%29"
              type="image/avif"
            />
            <source
              srcset="https://placekitten.com/408/287/m/328x328/filters:format%28webp%29:quality%2880%29"
              type="image/webp"
            />
            <img
              alt="some alt text"
              class="testimonial__image testimonial__imageMobile"
              data-nimg="1"
              decoding="async"
              height="328"
              loading="lazy"
              src="/_next/image?url=https%3A%2F%2Fplacekitten.com%2F408%2F287%2Fm%2F328x328%2Ffilters%3Aformat%2528jpeg%2529%3Aquality%252880%2529&w=750&q=75"
              srcset="/_next/image?url=https%3A%2F%2Fplacekitten.com%2F408%2F287%2Fm%2F328x328%2Ffilters%3Aformat%2528jpeg%2529%3Aquality%252880%2529&w=384&q=75 1x, /_next/image?url=https%3A%2F%2Fplacekitten.com%2F408%2F287%2Fm%2F328x328%2Ffilters%3Aformat%2528jpeg%2529%3Aquality%252880%2529&w=750&q=75 2x"
              style="color: transparent; max-width: 100%; height: auto;"
              width="328"
            />
          </picture>
          <p
            class="testimonial__details"
          >
            <span
              class="testimonial__name"
            >
              Test Name
            </span>
            <span
              class="testimonial__job"
            >
              Test Role
            </span>
          </p>
        </figcaption>
      </figure>
      <div
        class="testimonial__mainImageContainer"
      >
        <picture>
          <source
            srcset="https://placekitten.com/408/287/m/328x328/filters:format%28avif%29:quality%2880%29"
            type="image/avif"
          />
          <source
            srcset="https://placekitten.com/408/287/m/328x328/filters:format%28webp%29:quality%2880%29"
            type="image/webp"
          />
          <img
            alt="some alt text"
            class="testimonial__image"
            data-nimg="1"
            decoding="async"
            height="328"
            loading="lazy"
            src="/_next/image?url=https%3A%2F%2Fplacekitten.com%2F408%2F287%2Fm%2F328x328%2Ffilters%3Aformat%2528jpeg%2529%3Aquality%252880%2529&w=750&q=75"
            srcset="/_next/image?url=https%3A%2F%2Fplacekitten.com%2F408%2F287%2Fm%2F328x328%2Ffilters%3Aformat%2528jpeg%2529%3Aquality%252880%2529&w=384&q=75 1x, /_next/image?url=https%3A%2F%2Fplacekitten.com%2F408%2F287%2Fm%2F328x328%2Ffilters%3Aformat%2528jpeg%2529%3Aquality%252880%2529&w=750&q=75 2x"
            style="color: transparent; max-width: 100%; height: auto;"
            width="328"
          />
        </picture>
      </div>
    </div>
  </div>
</div>
`;
